In [3]:
import os
import sys
from os.path import join

import numpy as np
import scipy as sp


### Tutorial

In [4]:
from mvpa2.tutorial_suite import *

/anaconda2/lib/python2.7/site-packages/matplotlib/font_manager.py:279: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  'Matplotlib is building the font cache using fc-list. '
/anaconda2/lib/python2.7/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


Load full dataset (note: downloaded tutorial_data-0.4 from http://data.pymvpa.org/datasets/tutorial_data/, and placed in ./data/tutorial_data)

In [10]:
tutorial_data_path = '../data/tutorial_data/'
bold_fname = join(tutorial_data_path, 'sub001',
    'BOLD', 'task001_run001', 'bold.nii.gz')
ds = fmri_dataset(bold_fname)

 * Please note: warnings are printed only once, but underlying problem might occur many times *


In [12]:
print(ds.nfeatures)
print(ds.shape)

163840
(121, 163840)


Dataset has 121 samples (timepts, or vols), and 163840 features (voxels).

Now load the data, but mask it to only include voxels in the ventral temporal cortex

In [14]:
mask_fname = join(tutorial_data_path, 'sub001', 'masks', 'orig', 'vt.nii.gz')
ds = fmri_dataset(bold_fname, mask=mask_fname)

In [15]:
print(ds.nfeatures)
print(ds.shape)

577
(121, 577)


Only 577 features now, much better

In [20]:
#Check out the attributes present in this dataset
print(ds.a)
print(ds.sa)
print(ds.fa)

<DatasetAttributesCollection: imgaffine,imghdr,imgtype,mapper,voxel_dim,voxel_eldim>
<SampleAttributesCollection: time_coords,time_indices>
<FeatureAttributesCollection: voxel_indices>


In [21]:
ds.a.voxel_dim

(40, 64, 64)